Framework/DataHandling/src/SaveNexusGeometry.cpp

```diff
@@ -0,0 +1,57 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "MantidDataHandling/SaveNexusGeometry.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidGeometry/Instrument/InstrumentVisitor.h"
+#include "MantidNexusGeometry/NexusGeometrySave.h"
+
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/WorkspaceProperty.h"
+#include "MantidGeometry/Instrument/ComponentInfo.h"
+#include "MantidGeometry/Instrument/DetectorInfo.h"
+
+namespace Mantid {
+namespace DataHandling {
+using Mantid::API::WorkspaceProperty;
+using Mantid::Kernel::Direction;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(SaveNexusGeometry)
+
+//----------------------------------------------------------------------------------------------
+
+/** Initialize the algorithm's properties.
+ */
+void SaveNexusGeometry::init() {
+  const std::vector<std::string> exts{".nxs", ".hdf5"};
+  declareProperty(std::make_unique<WorkspaceProperty<API::MatrixWorkspace>>(
+                      "InputWorkspace", "", Direction::Input),
+                  "workspace containing the Instrument.");
+
+  declareProperty(std::make_unique<API::FileProperty>(
+                      "Filename", "", API::FileProperty::Save, exts),
+                  "full path save destination");
+
+  declareProperty("H5Path", "", "name of the H5 path save destination group.");
+}
+
+//----------------------------------------------------------------------------------------------
+/** Execute the algorithm.
+ */
+void SaveNexusGeometry::exec() {
+  API::MatrixWorkspace_const_sptr workspace = getProperty("InputWorkspace");
+  std::string destinationFile = getPropertyValue("FileName");
+  std::string rootFileName = getPropertyValue("H5Path");
+  const auto instrument = workspace->getInstrument();
+  auto instr = Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+  Mantid::NexusGeometry::NexusGeometrySave::saveInstrument(
+      instr, destinationFile, rootFileName);
```

Comment:
Don't forget progress reporting!

Reply:
am i missing a library other than #include <gmock/gmock.h>? 

these arent being recognised in vsual studio:

GNU_DIAG_OFF_SUGGEST_OVERRIDE
  MOCK_METHOD1(doReport, void(const std::string &));
  GNU_DIAG_OFF_SUGGEST_OVERRIDE

ive added the gmock libraries to cmakelists also in datahandlingtest

Reply:
You do not need gmock dependencies of any kind in `NexusGeometry` or `DataHandling` (only in the ...Test executables). You should be hooking into the Algorithm Report mechanism. See example [here](https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/src/Rebin.cpp#L186). You use an object of REAL type (not mock).

Reply:
I mean to add the mock progress base for the algorithm's own test suite. But guess I could skip it it since I have one In NexusGeometrySaveTest.

Reply:
That is tested already and would not be the responsibility of the testing for the `SaveNexusGeometryTest`, at this granularity, we should assume that all function used i.e. `saveInstrument` do the correct thing.

Reply:
As described by new issue. Reporting is NOT going to be handled in this PR.